README.md

```diff
@@ -659,4 +660,13 @@ internal final class SubClass: BaseClass {
 
 
 *Rationale* Subclassing in instances where the original class was not built to support subclasses can be a common source of bugs. Marking classes as `final`
-indicates that it was developed under the assumption that it would act on its own without regard for subclasses. 
+indicates that it was developed under the assumption that it would act on its own without regard for subclasses.
+
+### Documentation ###
+
+Well documented code is critical to help other developers understand what the code is doing. Every **public** and **internal** types should be documented.
+Additionally developers should annotate a private function if the behavior of it is not trivial using the regular comment syntax `//`.
```

Comment:
I want to explain that even private code should be commented if the developer thinks the implementation is not trivial, I can change to `private piece of code` if you think `function` is too specific.